### DISCUSSION TITLE
How to extract references from chapters/files to compile a complete list of references?

### DISCUSSION THREAD

**Discussion by drganghe** [quarto-dev/quarto-cli#7359] [Q&A]
### Description

I have prepared lectures for a class, each has its auto generated references at the end of the slides.

I hope to create a complete list of references for the class that combines references from each lecture to facilitate student learning, is there anyway to do it automatically?

For example the following lectures:
https://github.com/drganghe/energy-climate-policy/tree/main/lectures

Thank you for any tip on how to do it.

**Comment by mcanouil** [comment]
Why do you want to extract?
Use the bib file with all the references and use `nocite` to generate the full list (https://quarto.org/docs/authoring/footnotes-and-citations.html#including-uncited-items).

**Reply by drganghe** [reply]
Wow, this is a cool function! I will try it next time to generate a full list from bib file.

I want to create an organized list of readings/reference by lecture/chapter (qmd files) so students can follow. For now I have to copy the rendered reference lecture by lecture. I suspect there is an automatic or easier method to do it.

**Reply by mcanouil** [reply]
Maybe you could use the section bibliographie extension, see <https://github.com/pandoc-ext/section-bibliographies>.
